airbyte-integrations/connectors/source-notion/unit_tests/test_streams.py

```diff
@@ -167,3 +181,43 @@ def test_user_stream_handles_pagination_correctly(requests_mock):
     records = stream.read_records(sync_mode=SyncMode.full_refresh)
     records_length = sum(1 for _ in records)
     assert records_length == 220
+
+
+@pytest.mark.parametrize("stream,parent,status_code,expected_availability,expected_reason_substring", [
+    (Users, None, 403, False, "This is likely due to insufficient permissions for your Notion integration. "),
+    (Blocks, Pages, 403, False, "This is likely due to insufficient permissions for your Notion integration. "),
+    (Users, None, 200, True, None)
+])
+def test_403_error_handling(stream, parent, status_code, expected_availability, expected_reason_substring):
+    """
+    Test that availability strategy flags streams with 403 error as unavailable
+    and returns custom Notion integration message.
+    """
+
+    with patch(f'source_notion.streams.{stream.__name__}._send_request') as mock_send_request:
```

Comment:
I suggest you use `requests_mock` for cases like this. It is more readable and does not require patching magic methods. See example [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-gitlab/unit_tests/test_streams.py#L153)

Reply:
Gotcha, thanks for sharing the example. I've refactored the test as suggested 👍

Reply:
Thanks! Let's ship it